### DISCUSSION TITLE
Under windows, how to modify the name of the built onnxruntime.dll dynamic library？

### DISCUSSION THREAD

**Discussion by shuiyu486** [microsoft/onnxruntime#7431] [Other Q&A] [answered]
>  

In windows10, I want to modify the name of the built dynamic library to onnxruntime_dml.dll, but I don't see the relevant options in ".\build.bat -h".

>   

I used vs2019 to open the onnxruntime.sln generated by the build, and by modifying the properties of the onnxruntime project, I successfully changed the generated onnxruntime.dll to onnxruntime_dml.dll, but this method is very cumbersome.

>   

Is there an easier way? thanks for your help！

**Comment by liqunfu** [comment]
https://github.com/microsoft/onnxruntime/blob/1fa6d8fe1c1e41fcec15d63a57b3a0b5458b697b/cmake/CMakeLists.txt#L26

Maybe plus some other places

**Reply by shuiyu486** [reply]
thank.

**Comment by snnn** [comment, marked as answer]
In the line 
https://github.com/microsoft/onnxruntime/blob/master/cmake/onnxruntime.cmake#L36 

You need to change "onnxruntime_add_shared_library(onnxruntime" to "onnxruntime_add_shared_library(onnxruntime_dml" , and search all the cmake files and replace the reference places.

**Reply by shuiyu486** [reply]
thank.
